src/document_store.zig

```diff
@@ -84,7 +79,7 @@ pub fn openDocument(self: *DocumentStore, uri: []const u8, text: []const u8) !*H
     const duped_uri = try std.mem.dupe(self.allocator, u8, uri);
     errdefer self.allocator.free(duped_uri);
 
-    return self.newDocument(duped_uri, duped_text);
+    return try self.newDocument(duped_uri, duped_text);
```

Comment:
Is this a necessary change?  
I prefer the `return <maybe_erroring_expression>` instead of `return try`, is there some reasoning to choose the latter?

Reply:
yes. it adds a new line to the error stack trace.

Reply:
Ah good to know, thanks.  
Let's go ahead with this style instead then.